torch/testing/_internal/common_utils.py

```diff
@@ -1466,14 +1466,12 @@ def download_file(url, binary=True):
         warnings.warn(msg, RuntimeWarning)
         raise unittest.SkipTest(msg) from e
 
-
 def find_free_port():
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-    sock.bind(('localhost', 0))
-    sockname = sock.getsockname()
-    sock.close()
-    return sockname[1]
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sock:
```

Comment:
Might not be relevant for this particular PR, but there is still a race condition here. Once the socket gets closed at the end of this call, there is no guarantee that the returned port number will be available for later use. Another process can grab it in-between.

Another solution could be to reserve a pool of port numbers in advance and hand them out as requested.

Reply:
Yes good catch. This function is called with a retry (seen in create_tcp_store()) to handle this race condition, so the main fix in this PR was to move the existing tests under that retry which will fix the flakiness.